pkg/block/fetcher.go

```diff
@@ -130,7 +130,7 @@ type MetadataFetcher interface {
 }
 
 type MetadataFilter interface {
-	Filter(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, synced *extprom.TxGaugeVec, incompleteView bool) error
+	Filter(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, markers map[ulid.ULID]*metadata.DeletionMark, synced *extprom.TxGaugeVec, incompleteView bool) error
```

Comment:
do you think this is needed for the filter? Wonder if we make deletion mark first citizen in fetcher rather :thinking:

Reply:
Only one filter uses them. Alternative I considered was to pass fetcher to the filter, but that seems weird. Since fetcher already has all the markers, it can as well just pass them to filters.

Reply:
We can start with that indeed then